cartridges/int_algolia_sfra/cartridge/scripts/hooks/order/afterPOST.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
+
+var Site = require('dw/system/Site');
+var algoliaData = require('*/cartridge/scripts/algolia/lib/algoliaData');
+var jobHelper = require('*/cartridge/scripts/algolia/helper/jobHelper');
+var Logger = require('dw/system/Logger');
+var Status = require('dw/system/Status');
+
+
+exports.afterPOST = function (order) {
+    var Algolia_EnableRealTimeInventory = algoliaData.getPreference('EnableRealTimeInventory');
+    if (Algolia_EnableRealTimeInventory) {
+        var reindexHelper = require('*/cartridge/scripts/algolia/helper/reindexHelper');
+
+        try {
+            var algoliaOperations = [];
+            var shipments = order.getShipments();
+            for (var i = 0; i < shipments.length; i++) {
+                var shipment = shipments[i];
+                var plis = shipment.getProductLineItems();
+
+                // Process each product line item based on shipment type
+                if (shipment.custom.shipmentType === 'instore') {
+                    // Handle in-store pickup shipments
+                    for (let j = 0; j < plis.length; j++) {
+                        let pli = plis[j];
+                        let product = pli.product;
+                        let productOps = getProductData(product, ['storeAvailability']);
+
+                        productOps.forEach(function(productOp) {
+                            // If the product is not available in the store anymore
+                            if (productOp.body.storeAvailability.indexOf(shipment.custom.fromStoreId) === -1) {
+                                algoliaOperations.push(productOp);
+                            }
+                        });
+                    }
+                } else {
+                    // Handle standard shipments
+                    for (let j = 0; j < plis.length; j++) {
+                        let pli = plis[j];
+                        let product = pli.product;
+                        let productOps = getProductData(product, ['in_stock']);
+
+                        productOps.forEach(function(productOp) {
+                            // If the product is not available in the inventory anymore
+                            if (productOp.body.in_stock === false || productOp.body.in_stock === undefined) {
```

Comment:
This doesn't need to be checked multiple times. The `productOps` contain the records operations for the various locales, but the `in_stock` attribute will be the same for all.

Based on my comment above, I would:
- Check if the product is in stock using `productFilter.isInStock()`
- If not in stock, generate the appropriate AlgoliaOperations based on the `IndexOutOfStock` flag

Reply:
fixed according to your comments